src/deepsparse/utils/onnx.py

```diff
@@ -229,14 +229,31 @@ def override_onnx_input_shapes(
         input for input in all_inputs if input.name not in initializer_input_names
     ]
 
-    # Make sure that given input shapes can map to the ONNX model
-    assert len(external_inputs) == len(input_shapes)
-
     # Input shapes should be a list of lists, even if there is only one input
     assert all(isinstance(inp, list) for inp in input_shapes)
 
-    # Overwrite the shapes
+    # If there is a single input shape given and multiple inputs,
+    # duplicate for all inputs to apply the same shape
+    if len(input_shapes) == 1 and len(external_inputs) > 1:
+        for i in range(len(external_inputs) - 1):
```

Comment:
Why do we leave out the last external input here?

Reply:
This is trying to duplicate input_shapes to the same size as external_inputs, so i had it do this external_inputs-1 times. I'm not a fan of this loop so cleaner suggestions would be nice

Reply:
Maybe something like
`input_shapes.extend([input_shapes[0] for i in range(1, len(external_inputs))])` ?

Reply:
Cool, that works and I like it better